test/acceptance/tests/connect/connect_inject_namespaces_test.go

```diff
@@ -0,0 +1,147 @@
+package connect
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+)
+
+const staticServerNamespace = "ns1"
+const staticClientNamespace = "ns2"
+
+// Test that Connect works in a default installation
+func TestConnectInjectNamespaces(t *testing.T) {
+	cfg := suite.Config()
+	if !cfg.EnableEnterprise {
+		t.Skipf("skipping this test because -enable-enterprise is not set")
+	}
+
+	cases := []struct {
+		name                 string
+		destinationNamespace string
+		mirrorK8S            bool
+		secure               bool
+	}{
+		{
+			"single destination namespace (non-default)",
+			staticServerNamespace,
+			false,
+			false,
+		},
+		{
+			"single destination namespace (non-default); secure",
+			staticServerNamespace,
+			false,
+			true,
+		},
+		{
+			"mirror k8s namespaces",
+			staticServerNamespace,
+			true,
+			false,
+		},
+		{
+			"mirror k8s namespaces; secure",
+			staticServerNamespace,
+			true,
+			true,
+		},
+	}
```

Comment:
Do you want to make a note that we're not testing auto-encrypt? Presumably because the tests were too long?

Reply:
Sure, I'll make a note. I just didn't think there are any special changes we needed to do for auto-encrypt and namespaces, that's why I didn't add a test for it. I thought having just one secure installation was good enough.